Replace dynamic by Object/Object? in router.dart

aminedakhlii opened the pull request:
Changed all dynamic variables in router.dart to Object or Object? to enhance null safety by decreasing the usage of dynamic type in the flutter repository.  

Closes #77357

aminedakhlii committed:
replace dynamic to Object in router.dart

aminedakhlii commented:
@googlebot I signed it!

goderbauer commented:
Looks like the analyzer (https://github.com/flutter/flutter/pull/79209/checks?check_run_id=2210134619) and some existing tests are not happy with this change. Can you please take a look?

goderbauer commented:
Looks like the analyzer and some tests are still unhappy. Can you take another look?

engine-flutter-autoroll committed:
75997c53b Roll Skia from 3d358fe2e533 to f1126836f9ef (1 revision) (flutter/engine#25351) (#79531)

engine-flutter-autoroll committed:
4b52a9f14 Roll Skia from f1126836f9ef to 7f38a89654b2 (1 revision) (flutter/engine#25352) (#79538)

engine-flutter-autoroll committed:
Roll Engine from 4b52a9f148c6 to 686761833a6f (3 revisions) (#79544)

cbracken committed:
Correct misspellings throughout the codebase (#79500)

Piinks committed:
Add dart Fix for TextSelectionThemeData (#78897)

engine-flutter-autoroll committed:
9651d11fe Roll Dart SDK from 22dfe0dc39b1 to ff3caa6e6324 (1 revision) (flutter/engine#25356) (#79557)

jmagman committed:
Migrate build_info to null safety (#79469)

Commit:
[flutter_tools] use Url path.Context for joining URI (#79566)

Commit:
Mark integration_test_test as flaky (#79561)

engine-flutter-autoroll committed:
2edfc020b Roll Fuchsia Linux SDK from T_kNQ9dqg... to kafCPE8Hd... (flutter/engine#25358) (#79569)

engine-flutter-autoroll committed:
310fcc77f [google_maps_flutter] Fix NNBD migration mistake in example (flutter/plugins#3768) (#79570)

a14n committed:
remove unnecessary trailing commas (#79562)

engine-flutter-autoroll committed:
536742155 Roll Skia from 7f38a89654b2 to 87fc65710104 (4 revisions) (flutter/engine#25359) (#79576)

keyonghan committed:
Revert "Mark Mac_ios flutter_gallery__transition_perf_e2e_ios32 flaky (#79496)" (#79560)

This reverts commit dff957fb5ca833a16e511d638c5857b795816928.

Commit:
[flutter_tools] Move sksl writing out of vm_service (#79455)

engine-flutter-autoroll committed:
55b860083 [webview_flutter] Fix scroll bar position for Android non-hybrid WebViews (flutter/plugins#3765) (#79582)

jiahaog committed:
[flutter_tools] Normalize paths before checking for Integration Tests (#79521)

jason-simmons committed:
Restore support for the --local-engine flag in devicelab/bin/run.dart (#79583)

engine-flutter-autoroll committed:
Roll Engine from 536742155889 to 6b874395ee4a (3 revisions) (#79584)

engine-flutter-autoroll committed:
8531d5614 [video_player]Update README.me (flutter/plugins#3713) (#79586)

dannyvalentesonos committed:
Modified the computeMinIntrinsicHeight method to calculate the height properly (#75513)

dkwingsmt committed:
Reland Make LogicalKeyboardKey.keyLabel a getter and better #79100 (#79302)

engine-flutter-autoroll committed:
f85f47139 Roll Dart SDK from ff3caa6e6324 to e3e901344f72 (1 revision) (flutter/engine#25363) (#79589)

LongCatIsLooong committed:
Revert "Don't call onChanged callbacks when formatter rejects the change & handle text input formatter exceptions. (#78707)" (#79590)

jiahaog committed:
[flutter_tools] Fix breakage when test main is tagged with @doNotStore (#79501)

jmagman committed:
Migrate analysis to null safety (#79525)

jmagman committed:
Migrate fingerprint to null safety (#79587)

engine-flutter-autoroll committed:
Roll Engine from f85f47139a73 to 6e0ee453d026 (4 revisions) (#79600)

jiahaog committed:
Remove the warning about the Integration Test plugin (#79520)

srawlins committed:
Remove "unnecessary" imports. (#78666)

srawlins committed:
Remove "unnecessary" imports. (#78663)

mateusfccp committed:
Remove references to @required in favor of the  keyword (#79239)

corpulentcoffee committed:
Do not run version check if generating shell completion and stdout isn't terminal (#79418)

jmagman committed:
Migrate android_studio to null safety (#79504)

LongCatIsLooong committed:
`enterText` to move the caret to the end (#79506)

engine-flutter-autoroll committed:
Roll Engine from 6e0ee453d026 to 7636b14c895a (5 revisions) (#79602)

engine-flutter-autoroll committed:
Roll Engine from 7636b14c895a to bed3a4a7a016 (2 revisions) (#79606)

LongCatIsLooong committed:
Revert "`enterText` to move the caret to the end (#79506)" (#79654)

This reverts commit 67d8556baa9656b2e91c8dc886b86d5cf8d03cf1.

Commit:
Create dev/ci.yaml (#79396)

Commit:
[flutter_tools] split features into 2 libraries, migrate to null safety (#79603)

Commit:
Add removeListenerWhileNotifying benchmark for ChangeNotifier (#78161)

Commit:
[flutter_tools] consume package:process exceptions (#79657)

chunhtai committed:
Makes text selection match the native behavior (#79308)

canastro committed:
Support block delete with word and line modifiers (#77172)

jmagman committed:
Migrate android_sdk and application_package to null safety (#79611)

Piinks committed:
Automatically applying Scrollbars on desktop platforms with configurable ScrollBehaviors (#78588)

xu-baolin committed:
fix the position of popu menu when have unsage area (#78395)

trymnilsen committed:
Apply carriedVelocity unless substantially different (#79382)

engine-flutter-autoroll committed:
2d7bb8755 Roll Skia from 988b7043e54c to 42d753031d3f (1 revision) (flutter/engine#25374) (#79614)

jiahaog committed:
[flutter_tools] Remove exception for chain stack traces for Integration Tests (#79618)

gspencergoog committed:
Revert "Support block delete with word and line modifiers (#77172)" (#79674)

This reverts commit 75453e6ac6647e014765ae6d863e199efc152a16 because it breaks flutter/test/rendering/editable_test.dart

Landing on red to fix the build.

LongCatIsLooong committed:
Reland 79654 (#79656)

justinmc committed:
Fix typo: edition -> editing (#79668)

Just a docs typo fix.

engine-flutter-autoroll committed:
76a417ca8 [local_auth] Update Jetpack dependencies (flutter/plugins#3786) (#79667)

bkonyi committed:
[ Web ] Register service extensions with DDS, not DWDS (#79479)

engine-flutter-autoroll committed:
Roll Engine from 2d7bb8755f71 to 57b0853bcc4f (16 revisions) (#79675)

Hixie committed:
Fix a legacy TODO (#77454) (#79061)

Commit:
[flutter_tools] allow winuwp build to setup generated Cmake (#79676)

engine-flutter-autoroll committed:
Roll Engine from 57b0853bcc4f to 05b479d3b72c (3 revisions) (#79687)

engine-flutter-autoroll committed:
Roll Engine from 05b479d3b72c to a0b52ae6bf31 (3 revisions) (#79690)

engine-flutter-autoroll committed:
d90f9ab58 Roll Dart SDK from 624d14c61b2d to 0193522c6638 (1 revision) (flutter/engine#25405) (#79700)

a14n committed:
add trailing comma when argList is splitted (#79650)

engine-flutter-autoroll committed:
58215eb33 Roll Fuchsia Linux SDK from b6atUpuKK... to y_R-US0DL... (flutter/engine#25406) (#79703)

engine-flutter-autoroll committed:
9bc0f744e Eliminate error message for missing CMAP table (flutter/engine#25403) (#79713)

engine-flutter-autoroll committed:
e393800c6 Roll Dart SDK from 0193522c6638 to 0d7b8ec13047 (1 revision) (flutter/engine#25407) (#79717)

engine-flutter-autoroll committed:
a6b47deaa Roll Dart SDK from 0d7b8ec13047 to 00eaa3c60d31 (1 revision) (flutter/engine#25408) (#79725)

engine-flutter-autoroll committed:
Roll Engine from a6b47deaacca to 7a5165d1e2a9 (9 revisions) (#79768)

engine-flutter-autoroll committed:
3a53b28e9 Roll Fuchsia Mac SDK from 4WKVCYCPm... to F4eUNa2mh... (flutter/engine#25419) (#79777)

Hixie committed:
Clarify documentation for DragAnchorStrategy. (#79160)

jmagman committed:
Migrate doctor_validator to null safety (#79682)

SeongYunKim committed:
Fix typo in scroll_activity.dart (#79771)

engine-flutter-autoroll committed:
1412d3fc5 Roll Skia from 5c7ab08937ca to fd7252fa2392 (3 revisions) (flutter/engine#25422) (#79794)

mdebbar committed:
[web] Enable material text field tests on web (#79665)

jmagman committed:
Migrate tools chrome to null safety (#79801)

jmagman committed:
Migrate linux_doctor to null safety (#79804)

engine-flutter-autoroll committed:
4322497bf [google_maps_flutter] remove unnecessary test (flutter/plugins#3787) (#79802)

jmagman committed:
Migrate some workflows and doctor validators to null safety (#79807)

jmagman committed:
Migrate vscode to null safety (#79805)

engine-flutter-autoroll committed:
7aac2b301 Roll Skia from fd7252fa2392 to e296c56735c4 (4 revisions) (flutter/engine#25423) (#79803)

jcollins-g committed:
Update dartdoc to 0.42.0 (#79795)

xu-baolin committed:
improve the scrollbar behavior when viewport size changed (#76102)

Commit:
[flutter_tools] split cache into API and implementation (#79580)

Arkangel12 committed:
Fix error when choose noon on time picker (#78478)

engine-flutter-autoroll committed:
bd2dbb151 [tool] refactor publish plugin command (flutter/plugins#3779) (#79812)

engine-flutter-autoroll committed:
Roll Engine from 7aac2b301506 to 95913801a5a8 (2 revisions) (#79814)

darrenaustin committed:
Revert "Fix error when choose noon on time picker (#78478)" (#79828)

This reverts commit 6dd77cc4 as it is breaking the build.

mdebbar committed:
[web] Enable Cupertino text field tests on web (#79817)

engine-flutter-autoroll committed:
00ec3cdcc [google_maps_flutter] Reword README (flutter/plugins#3784) (#79820)

engine-flutter-autoroll committed:
d5fdd57ce Fix texture corruption on Windows (flutter/engine#25349) (#79821)

xu-baolin committed:
update CupertinoPicker.selectionOverlay to nullable (#79620)

engine-flutter-autoroll committed:
Roll Plugins from 00ec3cdccb3c to 4bd178b9fd89 (2 revisions) (#79837)

engine-flutter-autoroll committed:
Roll Engine from d5fdd57ce698 to e74b6497b803 (9 revisions) (#79839)

* 1d8d2df27 Rename EngineEmbedderApiModifier to EngineModifier (flutter/engine#25362)

* 057ed2288 Roll Skia from b14a819122ee to 0a12b85f160f (5 revisions) (flutter/engine#25430)

* 079857da6 Roll Fuchsia Linux SDK from b22xnKMfp... to txu1APnex... (flutter/engine#25431)

* 23ac479f0 Extract Dart logging to the embedders (flutter/engine#25402)

* 5444e2bac Remove the DartIsolate::PostMessage timeline event (flutter/engine#25433)

* 253a298ea Roll Skia from 0a12b85f160f to 974c82150204 (1 revision) (flutter/engine#25434)

* 61dca0fc8 Roll Dart SDK from c274ba72ec73 to 3dd1630ab4d4 (1 revision) (flutter/engine#25432)

* f09b189f5 Allow embedders to set a custom log tag (flutter/engine#25435)

* e74b6497b Roll Fuchsia Mac SDK from F4eUNa2mh... to IKx9cvrtJ... (flutter/engine#25436)

HansMuller committed:
Redo Removed accentColor dependency from ExpansionTile (#79829)

engine-flutter-autoroll committed:
016ee0cda Roll Dart SDK from 3dd1630ab4d4 to b55f7a132f67 (1 revision) (flutter/engine#25438) (#79845)

aminedakhlii commented:
@goderbauer I believe the analyzer is finally happy and all the tests are successful except for the Windows framework_tests_libraries for some reason, could you please help me figure out how to approach this issue and could you review the pull request.

engine-flutter-autoroll committed:
81aad8f25 [google_map_flutter] fix version (flutter/plugins#3790) (#79851)

engine-flutter-autoroll committed:
Roll Engine from 016ee0cdafdd to 829d03a03466 (5 revisions) (#79858)

douglasiacovelli committed:
Add back button listener widget (#79642)

engine-flutter-autoroll committed:
Roll Engine from 829d03a03466 to 3e59383e106e (2 revisions) (#79872)

jmagman committed:
Migrate intellij_validator to null safety (#79813)

jmagman committed:
Migrate android_workflow to null safety (#79806)

engine-flutter-autoroll committed:
Roll Engine from 3e59383e106e to 9166235f81db (2 revisions) (#79876)

srawlins committed:
Remove "unnecessary" imports in test/commands.shard. (#78665)

srawlins committed:
Remove "unnecessary" imports. (#78664)

chunhtai committed:
Fix RenderEditable register the raw keyboard listener twice (#79877)

* Fix RenderEditable register the raw keyboard listener twice

* skip web test

engine-flutter-autoroll committed:
Roll Engine from 9166235f81db to 4a1fd38dbc4a (3 revisions) (#79880)

Commit:
[flutter_tools] generate empty plugin registrants for winuwp and fix local engine for winuwp_ (#79827)

Commit:
[flutter_tools] migrate cache to null safety (#79864)

engine-flutter-autoroll committed:
Roll Engine from 4a1fd38dbc4a to 1a126ebdd079 (3 revisions) (#79891)

jmagman committed:
Update Podfile.lock modified timestamp when pod install is run (#78823)

engine-flutter-autoroll committed:
3f2e2acc8 Switch script/tools over to the new analysis options (flutter/plugins#3777) (#79890)

Commit:
Reland [devicelab] Migrate Gallery to BuildTestTask (#78361)

Commit:
Revert "Reland [devicelab] Migrate Gallery to BuildTestTask (#78361)" (#79903)

This reverts commit 0c9781e3dd043f4fa9f356611bff88535c37931b.

jmagman committed:
Move cache from globals to globals_null_migrated (#79900)

mdebbar committed:
[web] Fix url strategy null safety (#79888)

engine-flutter-autoroll committed:
Roll Engine from 1a126ebdd079 to 92f4dfb0484f (3 revisions) (#79902)

Abhishek01039 committed:
Change the code sample (#79743)

jmagman committed:
Allow null config values (#79917)

engine-flutter-autoroll committed:
97ac042f0 [flutter_webview] Fix `allowsInlineMediaPlayback` ignored on iOS (flutter/plugins#3791) (#79923)

jmagman committed:
Migrate test/src/common to null safety (#79907)

cenumi committed:
fix cupertino refresh control style (#79842)

xu-baolin committed:
update BorderTween nullable (#79919)

aminedakhlii commented:
@goderbauer Could you please review this ?

christopherfujino committed:
Manually roll engine to 56b13559807bf4cb83d16c659757d6ae4f209490 (#79953)

56b135598 Roll Dart SDK from e778310fcdcc to dda1a31495e3 (1 revision) (#25470)
7ca642c80 Roll Skia from 0195bec798a9 to 2f5a91c531de (1 revision) (#25469)
5cea3a738 Refactor GPU access switch (#24356)
e2791be97 Roll Fuchsia Mac SDK from Z-pl5laFz... to a9NOB6sdo... (#25468)
4cd3ab8d8 Roll Fuchsia Linux SDK from dl1JUKVuj... to 0Db2pEb0U... (#25467)
3eb672c77 Roll Dart SDK from 9704370488b0 to e778310fcdcc (1 revision) (#25466)
e1b291c9b Roll Skia from d986f476ae68 to 0195bec798a9 (1 revision) (#25463)
186eeb752 Revert "Reland: Call Dart plugin registrant if available (#25375)" (#25462)
63fda233d Roll Skia from 601abfacc736 to d986f476ae68 (5 revisions) (#25461)
c70b8ca59 Roll Dart SDK from b9471aadaa1a to 9704370488b0 (1 revision) (#25459)

canastro committed:
Support block delete with word and line modifiers (#79695)

Support for keyboard backspace/delete shortcuts

jmagman committed:
Revert "Support block delete with word and line modifiers (#79695)" (#79968)

This reverts commit 0ba2f6a87fbfd857c0a4e01a47e824dcab5462de.

maheshj01 committed:
docs: add an example to ReorderableListView.builder (#79193)

docs: add an example to ReorderableListView.builder.

justinmc committed:
Revert "Fix a legacy TODO (#77454) (#79061)" (#79959)

This reverts commit b070ed3c04caa6e348c068b1fe8396527fc46554.

campovski committed:
Dedup CupertinoAlertDialog and CupertinoActionSheet source code (#78202)

jmagman committed:
Remove MockStdIn and MockStream (#78276)

christopherfujino committed:
migrate empty exceptions to tool exit with error message (#79898)

engine-flutter-autoroll committed:
1ef73d9c4 [local_auth] Unnecessary reassignment in example removed (flutter/plugins#2983) (#79954)

jmagman committed:
Mark integration_test_test as unflaky (#79961)

canastro committed:
Support block delete with word and line modifiers (#79973)

Adds support for overridable delete/backspace keyboard shortcuts

engine-flutter-autoroll committed:
e46471e1e Roll Dart SDK from dda1a31495e3 to 3f36938a7cff (2 revisions) (flutter/engine#25472) (#79971)

jason-simmons committed:
Fix an NNBD error in the Gallery text form field demo (#79816)

jmagman committed:
Start migrating flutter_tools test src to null safety (#79908)

christopherfujino committed:
add new try/prod builders for validate_ci_config (#79913)

engine-flutter-autoroll committed:
d5ebaa536 [in_app_purchase] Configured example app to use StoreKit Testing on iOS 14 (flutter/plugins#3772) (#79978)

engine-flutter-autoroll committed:
cf8682397 Roll Skia from 2f5a91c531de to 5c6258287461 (3 revisions) (flutter/engine#25473) (#79979)

srawlins committed:
Remove "unnecessary" imports in test/widgets (#79608)

goderbauer commented:
Looks like a flake, I restarted the test, let's see if it passes this time.

HansMuller committed:
Redo fix for button.icon layout overflow (#79085)

aminedakhlii commented:
@goderbauer it did!

jmagman committed:
Migrate tool version to null safety (#79911)

goderbauer commented on packages/flutter/lib/src/widgets/router.dart:
```diff
@@ -272,7 +272,7 @@ class Router<T> extends StatefulWidget {
   /// Since this delegate is the primary consumer of the [routeInformationProvider],
   /// it must not be null if [routeInformationProvider] is not null.
-  final RouteInformationParser<T>? routeInformationParser;
+  final RouteInformationParser<Object>? routeInformationParser;
```
Why this changing from T to Object?

goderbauer replied:
(here and elsewhere in the PR)

aminedakhlii replied:
@goderbauer Because I changed RouteInformationParser definition from RouteInformationParser<T> to RouteInformationParser<Object> and T is nullable so it won't be compatible and I had to change it here. 
Same goes for all changes from T to Object or Object? elsewhere

goderbauer commented on packages/flutter/lib/src/widgets/router.dart:
```diff
@@ -334,7 +334,7 @@ class Router<T> extends StatefulWidget {
   ///  * [of], a similar method that returns a non-nullable value, and will
   ///    throw if no [Router] ancestor exists.
-  static Router<dynamic>? maybeOf(BuildContext context) {
+  static Router<Object>? maybeOf(BuildContext context) {
```
This should probably be typed similar to how its done for `ModalRoute.of`

goderbauer replied:
(same for the `of` method above.)

goderbauer commented on packages/flutter/lib/src/widgets/router.dart:
```diff
@@ -619,23 +619,24 @@ class _RouterState<T> extends State<Router<T>> {
   }
 
-  static final Future<dynamic> _never = Completer<dynamic>().future; // won't ever complete
+  static final Future<Object> _never = Completer<Object>().future; // won't ever complete
```
this could be `void`

goderbauer commented on packages/flutter/lib/src/widgets/router.dart:
```diff
@@ -619,23 +619,24 @@ class _RouterState<T> extends State<Router<T>> {
 
   _AsyncPassthrough<void> _verifyRouterDelegatePushStillCurrent(Object? transaction, Router<T> originalWidget) {
     return (void data) {
+
```
remove blank line

goderbauer commented on packages/flutter/lib/src/widgets/router.dart:
```diff
@@ -678,7 +679,7 @@ class _RouterState<T> extends State<Router<T>> {
       routeInformationParser: widget.routeInformationParser,
       routerDelegate: widget.routerDelegate,
-      routerState: this,
+      routerState: this as _RouterState<Object>,
```
this doesn't look right

goderbauer commented on packages/flutter/lib/src/widgets/router.dart:
```diff
@@ -704,9 +705,9 @@ class _RouterScope extends InheritedWidget {
-  final RouteInformationParser<dynamic>? routeInformationParser;
-  final RouterDelegate<dynamic> routerDelegate;
-  final _RouterState<dynamic> routerState;
+  final RouteInformationParser<Object>? routeInformationParser;
+  final RouterDelegate<Object> routerDelegate;
+  final _RouterState<Object> routerState;
```
These probably have to be `Object?`

jmagman committed:
Use testUsingContext in downgrade_upgrade_integration_test (#79975)

jmagman committed:
Remove flutterNext (#79989)

xu-baolin committed:
Fix a rotation gesture bug (#79035)

srawlins committed:
Remove "unnecessary" imports. (#79610)

dkwingsmt committed:
Revert "Remove "unnecessary" imports. (#79610)" (#80070)

This reverts commit 98a96189937105f51e58e65543e2c6019a82cb33.

aminedakhlii commented:
@goderbauer Could you check again ?

godofredoc committed:
Disable gradle_plugin_light_apk_test. (#80072)

godofredoc committed:
Disable win_gradle_plugin_light_apk_test. (#80083)

* Disable win_gradle_plugin_light_apk_test.

Bug: https://github.com/flutter/flutter/issues/80056

* Enable the mac version.

jmagman committed:
Convert some general.shard base tests to null safety (#79985)

jmagman committed:
Cast config values to dynamic values instead of Object (#80011)

jmagman committed:
Migrate intellij tests and context_test to null safety (#80008)

jmagman committed:
Migrate more tool unit tests to null safety (#80002)

jmagman committed:
Migrate template to null safety (#80016)

jmagman committed:
Migrate fake_process_manager to null safety (#80018)

chingjun committed:
Expose `packagePath` as a getter in FlutterCommand. (#80026)

Commit:
[flutter_tools] handle out of date xcode config in assemble (#79983)

Commit:
[flutter_tools] Allow flutter build aar to know the null safety mode ahead of time (#80000)

Commit:
[flutter_tools] add timeline ANR integration test (#79991)

xu-baolin committed:
fix paste crash when section is invalid (#78948)

justinmc committed:
Reland InteractiveViewer.builder (#79287)

feHouyushan committed:
When updating TabController widget, if _controller.index >= widget.length, update _animationController's value (#79535)

engine-flutter-autoroll committed:
Roll Engine from cf86823971a8 to 57a0fbb77298 (14 revisions) (#80080)

johnpryan committed:
Recommend an SDK dependency in integration_test README (#79591)

godofredoc committed:
Update the no-response template. (#79925)

goderbauer committed:
Reduce severity of memory leak when BuildContext/Element is retained (#79957)

goderbauer committed:
Fix doc for FlexColumnWidth.value (#79990)

goderbauer committed:
Clear listeners when AnimationController is disposed (#79998)

Abhishek01039 committed:
fix type in shader_warm (#80028)

Piinks committed:
Revert "improve the scrollbar behavior when viewport size changed (#76102)" (#80059)

engine-flutter-autoroll committed:
89ccc0ead Fix and update version checks (flutter/plugins#3792) (#80077)

jmagman committed:
Check if reg can run before calling it on Windows (#79905)

engine-flutter-autoroll committed:
Roll Engine from 57a0fbb77298 to 994531d3d9ad (4 revisions) (#80090)

jmagman committed:
Migrate persistent_tool_state to null safety (#80085)

jmagman committed:
Migrate visual_studio_test (#80088)

engine-flutter-autoroll committed:
Roll Engine from 994531d3d9ad to a9f4883dc8ad (3 revisions) (#80093)

jmagman committed:
Migrate web_validator to null safety (#80095)

jmagman committed:
Tell user how to enable available but disabled features (#79977)

HansMuller committed:
Added ButtonStyle.maximumSize (#80087)

engine-flutter-autoroll committed:
1dbc26232 [local_auth] Fix iOS crash when no localizedReason (flutter/plugins#3780) (#80118)

atrope committed:
Catch errors during android plugin registration (#78964)

Commit:
[flutter_tools] cache for UWP artifacts (#80069)

engine-flutter-autoroll committed:
ee5d15e44 [in_app_purchase] Added currency code and numerical price to product detail model. (flutter/plugins#3794) (#80124)

engine-flutter-autoroll committed:
Roll Engine from a9f4883dc8ad to a7777cdf54b4 (3 revisions) (#80135)

jmagman committed:
Migrate first_run and bot_detector to null safety (#80096)

Commit:
[flutter_tools] treat win32 plugins as uwp plugins (#80131)

justinmc committed:
Add a potentially missing TextSelectionOverlay dispose (#79988)

jmagman committed:
Remove crash_reporting and github_template from reporting library (#80139)

aminedakhlii commented:
@goderbauer I believe I fixed everything you mentioned in the review, could you review the last commit please?

engine-flutter-autoroll committed:
Roll Engine from a7777cdf54b4 to 29d9070407cc (3 revisions) (#80152)

jmagman committed:
Mark mac_gradle_plugin_light_apk_test flaky (#80163)

Commit:
[flutter_tools] dont require a device for screenshot type skia/rasterizer (#80153)

jmagman committed:
Migrate tools test fakes to null safety (#80154)

engine-flutter-autoroll committed:
Roll Engine from 29d9070407cc to c34c490cac42 (2 revisions) (#80165)

engine-flutter-autoroll committed:
742dac8b7 Roll Skia from 3cffe81f0d16 to d3a1df8da790 (7 revisions) (flutter/engine#25519) (#80169)

engine-flutter-autoroll committed:
Roll Engine from 742dac8b796e to 398bede5b914 (2 revisions) (#80170)

engine-flutter-autoroll committed:
8863afff1 [web] Use greatest span font size on the parent paragraph (flutter/engine#25513) (#80173)

chinmoy12c committed:
Added enableFeedback property to FloatingActionButton (#69826)

lrhn committed:
Change test to not implement ByteData (#80049)

jmagman committed:
Migrate features_tests and other tool tests to null safety (#80168)

jmagman committed:
Reduce number of project creates in gradle_plugin_*_apk_tests (#80171)

HansMuller committed:
Added MaterialState.scrolledUnder and support in AppBar.backgroundColor (#79999)

darrenaustin committed:
Make the ReorderableListView padding scroll with the list. (#80251)

jmagman committed:
Remove flutter_command dependency from reporting library (#80159)

jmagman committed:
Cache location of Java binary in devicelab tests (#80172)

HansMuller committed:
Fixed typos in AppBar.toolbarHeight API doc sample (#80316)

Commit:
[flutter_tools] symlink win32 plugins as UWP plugins (#80242)

gspencergoog committed:
Remove a dynamic that is no longer necessary (and the TODO for it) (#80294)

Commit:
make pub super verbose for LUCI CI (#80319)

jmagman committed:
Revert "Remove a dynamic that is no longer necessary (and the TODO for it) (#80294)" (#80326)

This reverts commit 12ec7dc8790feb437566c8b8e6b6e883a110b4d1.

angjieli committed:
Revert "Reland InteractiveViewer.builder (#79287)" (#80305)

This reverts commit 0fd75528de3f8d5225ce721e8be27041b8f6e2b6.

HansMuller committed:
Revert "Added MaterialState.scrolledUnder and support in AppBar.backgroundColor (#79999)" (#80380)

This reverts commit 17870d5d7eb81bc2021454093e9115c2c0d3c850.

engine-flutter-autoroll committed:
Roll Engine from 8863afff1650 to 9b6c3d4174ef (3 revisions) (#80183)

jmagman committed:
Migrate flutter_tool plugins.dart to null safety (#80304)

jmagman committed:
Mark linux_validate_ci_config not flaky (#80315)

jmagman committed:
Remove mocks from fucshia_pm_test, reduce createMockProcess usage (#80321)

jmagman committed:
Mark Mac gradle_plugin_light_apk_test not flaky (#80325)

jmagman committed:
Move gradle_non_android_plugin_test from its own test builders into tools integration.shard (#80161)

miDeb committed:
Calculate the system overlay style based on the AppBar background color (#75091)

xu-baolin committed:
update SearchDelegate's leading and actions widgets can be null (#80006)

royarg02 committed:
[flutter_tools] Show upstream remote in `flutter doctor -v` (#80101)

Commit:
[flutter_tools] fix null check in crash reporter (#80382)

dnfield committed:
Add an option to specify the output for code size intermediate files (#80310)

Abhishek01039 committed:
removed cirrus (#78761)

pedromassango committed:
Add TextOverflow into TextStyle (#78927)

jmagman committed:
Pull XCDevice out of xcode.dart (#80324)

engine-flutter-autoroll committed:
f34add38d [image_picker_platform_interface] Added pickMultiImage (flutter/plugins#3782) (#80344)

engine-flutter-autoroll committed:
Roll Engine from 9b6c3d4174ef to 04cd0db03201 (25 revisions) (#80390)

dnfield committed:
roll packages (#80307)

gspencergoog committed:
Reland "Remove a dynamic that is no longer necessary (and the TODO for it) (#80294)" (#80373)

cmc5788 committed:
Add --dart-define to build bundle (#80333)

pedromassango committed:
Add onTap and autocorrect into CupertinoSearchTextField (#79966)

Commit:
Autocomplete and RawAutocomplete initialValue parameter (#80257)

Abhishek01039 committed:
Add timer picker link (#80336)

HansMuller committed:
Re-land "Added MaterialState.scrolledUnder and support in AppBar.backgroundColor" (#80395)

engine-flutter-autoroll committed:
b548ae186 [google_sign_in] Fix "pick account" on iOS (flutter/plugins#3805) (#80397)

engine-flutter-autoroll committed:
Roll Engine from 04cd0db03201 to dade70a05f6a (5 revisions) (#80398)

HelloCore committed:
Include debug symbol in xcframework when building iOS-framework (#80065)

Commit:
[devicelab] Refresh documentation (#80064)

engine-flutter-autoroll committed:
Roll Engine from dade70a05f6a to 13d8565cf2e4 (4 revisions) (#80443)

christopherfujino committed:
skip roll_dev_integration_test (#80464)

goderbauer commented on packages/flutter/lib/src/widgets/router.dart:
```diff
@@ -306,7 +306,7 @@ class Router<T> extends StatefulWidget {
   ///  * [maybeOf], which is a similar function, but it will return null instead
   ///    of throwing an exception if no [Router] ancestor exists.
-  static Router<dynamic> of(BuildContext context) {
+  static Router<T> of <T extends Object?>(BuildContext context) {
```
remove space before <

goderbauer commented on packages/flutter/lib/src/widgets/router.dart:
```diff
@@ -334,9 +334,9 @@ class Router<T> extends StatefulWidget {
   ///  * [of], a similar method that returns a non-nullable value, and will
   ///    throw if no [Router] ancestor exists.
-  static Router<dynamic>? maybeOf(BuildContext context) {
+  static Router<T>? maybeOf <T extends Object?> (BuildContext context) {
```
remove space before < and after >

goderbauer commented on packages/flutter/lib/src/widgets/router.dart:
```diff
@@ -619,7 +619,7 @@ class _RouterState<T> extends State<Router<T>> {
   }
 
-  static final Future<dynamic> _never = Completer<dynamic>().future; // won't ever complete
+  static final Future<void> _never = Completer<void>().future; // won't ever complete
```
I think this one actually has to be dynamic because we cast it to all kinds of different types below.

aminedakhlii commented:
@goderbauer  I fixed what you mentioned again but for some reason the framework_tests-misc are failing, could you check why is that happening and review again?

goderbauer commented:
That test has been disabled, can you rebase on the latest master to pick up that change?

letsar committed:
Improve the performances of ChangeNotifier (#71947)

Commit:
[State Restoration] Restorable FormField and TextFormField (#78835)

* Restorable FormField and TextFormField

jmagman committed:
Split out XcodeProjectInterpreter from xcode_build_settings (#80399)

jmagman committed:
Add fourth tool integration subshard (#80460)

gspencergoog committed:
Handle null primary focus when receiving key events (#80302)

This allows the key handling code in the shortcut manager to function when there is a null primaryFocus. The primary focus can be null only in some special circumstances (e.g. when an app hasn't set any focus at all), but that shouldn't cause a crash.

kalebhermes committed:
[flutter_driver] Add waitForTappable to flutter_driver (#79581)

Piinks committed:
Fix FlexibleSpaceBar Opacity when AppBar.toolbarHeight > ktoolbarHeight (#80453)

HansMuller committed:
Updated the OutlinedButton class API doc (#80454)

HansMuller committed:
Added support for AppBarTheme.toolbarHeight (#80467)

engine-flutter-autoroll committed:
Roll Engine from 13d8565cf2e4 to 1a9cd85bc5dd (3 revisions) (#80470)
